docs/03-language-guide/14-equality.md

```diff
@@ -1,6 +1,131 @@
 ---
 title: Equality
 id: equality
+description: A reference guide to equality in Wing.
+keywords: [Winglang equality, Wing equality, equals, identity]
 ---
 
-TODO (for now, see the [language spec RFC](../999-contributors/999-rfcs/2023-06-12-language-spec.md))
+Checking for equality is performed with the `==` operator in Wing. It returns `true` if the two values are equal, and `false` otherwise.
+
+The main difference between equality in JavaScript and Wing is that `==` in Wing is not allowed to compare values of different types. For example, `1 == "1"` is not allowed in Wing, and will result in a compile-time error.
+
+The execution phase ([preflight or inflight](../02-core-concepts/01-preflight-and-inflight.md)) that a value was created in does not affect its equality. For example, a value created in preflight can be equal to a value created in inflight.
+
+Equality in Wing is a symmetric and transitive relationship - that is, (1) if `a == b`, then `b == a`, and (2) if `a == b` and `b == c`, then `a == c`.
+
+The following is a set of rules for checking equality:
+
+## Basic types
+
+1. Two `str` values are equal if they have the same characters in the same order.
+2. Two `num` values are equal if they have the same floating-point value. `NaN` is not equal to any value, including itself.
```

Comment:
Should we reference https://en.wikipedia.org/wiki/IEEE_754? Assuming we're gonna go by that (as js does).
That would include specifying that -0 == +0.